src/main/java/com/michelin/ns4kafka/service/StreamService.java

```diff
@@ -39,6 +42,21 @@ public List<KafkaStream> findAllForNamespace(Namespace namespace) {
             .toList();
     }
 
+    /**
+     * Find all Kafka Streams of a given namespace, filtered by given parameters.
+     *
+     * @param namespace The namespace
+     * @return A list of Kafka Streams
+     */
+    public List<KafkaStream> findAllForNamespace(Namespace namespace, KafkaStreamFilterParams params) {
```

Comment:
This method is basically the same than the `findByName` method, except it accepts regex special chars. 

I think you can safely update the `findByName` method to accept wildcards in the `String stream` parameter. The method will now return a List<KafkaStream>.

Reply:
I think the findByName is used for the endpoint /streams/{stream} which is then used by old clients.

Reply:
As seen together:
- I'll keep the findByName implementation since we prefer that old clients use the endpoint `/streams/{name}` which uses the `findByName` method and wildcard will not be supported for them: any wildcard will result in "not found".
- I'll rename the `findAllForNamespace` method to `findByWildcardName` which accepts wildcard names, used for filter.
- I'll put the `@Deprecated` annotation in the controller method corresponding to the `/streams/{name}` endpoint.